### ISSUE TITLE
TF Ranking Paper: Drive Quick Access NDCG Eval

### ISSUE THREAD

**Issue by eggie5** [tensorflow/ranking#89]
From what I understand of the Drive Quick Access Example presented in the TF Ranking paper and in [1]:

* binary relevance labels (1 click/query)
* listwise softmax loss (list_size~=4)
* pointwise scoring (group_size=1)
* Evaluation: MRR, ARP and _NDCG_?

Assuming all that above is correct, and given the binary relevance labels, how can one use a DCG metric? 

For binary relevance, Precision, Recall, AP, mAP, F1, MRR, and X@k are all applicable (because in this case, we only know which document is relevant instead of the degree of relevance). 

It's not meaningful to use nDCG to evaluate binary labels (because information gain always equals to 1 or 0).

[1] Quick Access: Building a smart experience for Google Drive

**Comment by ramakumar1729** [comment]
NDCG does make sense for single click scenarios. The value of NDCG would be $1/log_2(1 + r)$, where $r$ is the rank of the clicked document, as per the scores.

**Comment by bendersky** [comment]
Alex,
As discussed offline, in the NDCG with binary labels information gain does not change, but the position discount does, so the clicks at lower positions will lead to lower NDCG values. For one-click scenarios, NDCG should be related to MRR but with less steep position discount $1/log(1+r)$ v. $1/r$ (and non-zero scores only down to position $k$, for $NDCG@k$).

Closing this bug for now, as the metrics works as intended.

**Comment by LiangqunLu** [comment]
For the ndcg calculation, I have one problem. If there are many queries without positive relevance, will they be counted in DCG and NDCG? And if the documents of queries have fewer than the NDCG@k, does it mean NDCG@k = 0? Sorry that I have not figured out how the NDCG at TFR is calculated yet.
